client/components/fixed-navigation-header/index.tsx

```diff
@@ -55,16 +56,53 @@ interface Props {
 	className?: string;
 	children?: ReactNode;
 	navigationItems: { label: string; href?: string }[];
+	contentRef?: React.RefObject< HTMLElement >;
 }
 
 const FixedNavigationHeader: React.FunctionComponent< Props > = ( props ) => {
-	const { id, className, children, navigationItems } = props;
+	const { id, className, children, navigationItems, contentRef } = props;
+	const actionsRef = useRef< HTMLDivElement >( null );
+	const headerRef = useRef< HTMLElement >( null );
+
+	useEffect( () => {
+		if ( ! contentRef ) {
+			return;
+		}
+
+		const handleScroll = () => {
+			const headerHeight = headerRef?.current?.getBoundingClientRect().height;
+			const offset =
+				contentRef.current && headerHeight ? contentRef.current.offsetTop - headerHeight : 0;
+			const scrollPosition = window.scrollY;
+			const actionElement = actionsRef?.current;
+
+			if ( ! actionElement ) {
+				return;
+			}
+
+			if ( offset > 0 && scrollPosition < offset ) {
+				actionElement.style.visibility = 'hidden';
+			} else {
+				actionElement.style.visibility = 'visible';
+			}
+		};
+
+		window.addEventListener( 'scroll', handleScroll );
+		return () => {
+			window.removeEventListener( 'scroll', handleScroll );
+		};
+	}, [ contentRef ] );
 
 	return (
-		<Header id={ id } className={ className }>
+		<Header id={ id } className={ className } ref={ headerRef }>
 			<Container>
 				<Breadcrumb items={ navigationItems } />
-				<ActionsContainer>{ children }</ActionsContainer>
+				<ActionsContainer
+					ref={ actionsRef }
+					style={ { visibility: `${ contentRef ? 'hidden' : 'visible' }` } }
```

Comment:
Could having this property managed by a prop and then manipulated via the DOM API in `handleScroll` cause some problems? It seems ok, but I wondered if it might be safer to trigger `handleScroll` in the `useEffect` to set the initial value.

Also, I'm not sure if it's working as intended. When I scroll nothing seems to get hidden, or become visible. I think the intention is that the button at the top is hidden until the button in the main content is scrolled out of view.

https://user-images.githubusercontent.com/96462/141977229-0ee64b6b-003c-451a-bc4c-d3a7fe583d75.mov

Reply:
> Could having this property managed by a prop and then manipulated via the DOM API in `handleScroll` cause some problems? It seems ok, but I wondered if it might be safer to trigger `handleScroll` in the `useEffect` to set the initial value.

That makes more sense 👍 I made the change.

> Also, I'm not sure if it's working as intended. When I scroll nothing seems to get hidden, or become visible. I think the intention is that the button at the top is hidden until the button in the main content is scrolled out of view.
> Screen.Recording.2021-11-16.at.11.25.45.mov

I think it's working correctly now:

https://user-images.githubusercontent.com/1689238/142062714-632eca04-aa7e-4279-8d21-8b4ff3dff950.mov